apps/src/netsim/NetSimLogEntry.js

```diff
@@ -169,3 +174,41 @@ NetSimLogEntry.prototype.getLocalizedStatus = function () {
   }
   return '';
 };
+
+/**
+ * @returns {string} Localized "X of Y" packet count info for this entry.
+ */
+NetSimLogEntry.prototype.getLocalizedPacketInfo = function () {
+  return i18n.xOfYPackets({
+    x: this.getHeaderField(Packet.HeaderType.PACKET_INDEX),
+    y: this.getHeaderField(Packet.HeaderType.PACKET_COUNT)
+  });
+};
+
+/**
+ * @returns {string} 12-hour short time
+ */
+NetSimLogEntry.prototype.getTimeString = function () {
+  var date = new Date(this.timestamp);
+  var hours = date.getHours() % 12;
+  var hourString = (hours === 0 ? '12' : hours.toString());
+  var minuteString = ('0' + date.getMinutes()).substr(-2);
+  return hourString + ':' + minuteString;
```

Comment:
I've used this library in personal projects and liked it. Almost
recommended it myself
On Jun 2, 2015 5:03 PM, "Brad Buchanan" <notifications@github.com> wrote:

> In apps/src/netsim/NetSimLogEntry.js
> <https://github.com/code-dot-org/code-dot-org/pull/2568#discussion_r31583663>
> :
>
> > +NetSimLogEntry.prototype.getLocalizedPacketInfo = function () {
> > +  return i18n.xOfYPackets({
> > +    x: this.getHeaderField(Packet.HeaderType.PACKET_INDEX),
> > +    y: this.getHeaderField(Packet.HeaderType.PACKET_COUNT)
> > +  });
> > +};
> > +
> > +/**
> > + * @returns {string} 12-hour short time
> > + */
> > +NetSimLogEntry.prototype.getTimeString = function () {
> > +  var date = new Date(this.timestamp);
> > +  var hours = date.getHours() % 12;
> > +  var hourString = (hours === 0 ? '12' : hours.toString());
> > +  var minuteString = ('0' + date.getMinutes()).substr(-2);
> > +  return hourString + ':' + minuteString;
>
> I wonder if I should just pull down http://momentjs.com/ and use that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/2568/files#r31583663>.
>